### PR TITLE
Add Iam resources to LoginNodes

### PR DESCRIPTION
**Pull request by hehe7318**
### Description of changes
* Modify login_nodes_stack.py
  * add a function to add login node iam resources
  * add instance profile to LoginNodes Pool Launch Template
* Modify cdk_builder_utils.py
  * add CloudWatchAgentServerPolicy (AWS managed) and parallelcluster (Customer inline) policies to LoginNodes
* Modify test_cluster_stack.py
  * add unit test for LoginNodes IAM role and IAM policies

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

### CODE REVIEW

File: cli/src/pcluster/templates/cdk_builder_utils.py [hunk 1]
```diff
@@ -837,6 +837,56 @@ def _generate_head_node_pass_role_resources(self):
...
+                    "logs:CreateLogGroup",
+                ],
+                effect=iam.Effect.ALLOW,
+                resources=["*"],
+            ),
+            iam.PolicyStatement(
+                actions=["ssm:GetParameter"],
+                effect=iam.Effect.ALLOW,
+                resources=["arn:aws:ssm:*:*:parameter/AmazonCloudWatch-*"],
+            ),
```
Comment (maurizp) [inline cli/src/pcluster/templates/cdk_builder_utils.py]: This seems to be a duplicated of the `CloudWatchAgentServerPolicy`: why do you need to duplicate it?

**Review by maurizp — CHANGES_REQUESTED** [review]

File: cli/src/pcluster/templates/login_nodes_stack.py
Comment (maurizp) [inline cli/src/pcluster/templates/login_nodes_stack.py:1]: how did you test it?

File: cli/src/pcluster/templates/cdk_builder_utils.py (hunk 1 above)
Comment (hehe7318) [reply cli/src/pcluster/templates/cdk_builder_utils.py]: I have removed the duplicate policies. Thank you for reminding.

File: cli/src/pcluster/templates/login_nodes_stack.py
Comment (hehe7318) [reply cli/src/pcluster/templates/login_nodes_stack.py:1]: I tested by creating a test cluster and looked into the role to see if the policies were included. Should I create a unit test?

Comment (hehe7318) [reply cli/src/pcluster/templates/login_nodes_stack.py:1]: I have added unit tests for it.

**Review by maurizp — APPROVED** [review]

**Pull request merged**